<!DOCTYPE html>
<html>
<head>
    <title>Frequently asked questions</title>
    <meta http-equiv="Refresh" content="10">
    <link href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous">
    <link href="https://fonts.googleapis.com/css?family=Roboto&display=swap" rel="stylesheet">
    <style type="text/css">
        body {
            font-family: 'Roboto', sans-serif;
        }
        p {
            margin-bottom: 30px;
            text-align: justify;
        }
    </style>
</head>
<body>
    <div class="container">
        <center>
            <a href="/"><img width="500px" src="{{ url_for('static', filename='img/logo.png') }}"></a><br>
            <a href="/">&lt;&lt;&nbsp;Back</a>
            <h3>Frequently asked questions</h3>
        </center>
        <div class="row">
            <div class="col-md-1"></div>
            <div class="col-md-10">

                <h4><b>What are the sources that can be predicted?</b></h4>
<p>To date, FORENSIC holds the fecal signatures of 8 sources for the V6 and V4V5 regions of the 16S rRNA gene. It includes the cat, dog, pet (= cat + dog), cow, deer, ruminant (= cow + deer), pig, sewage.</p>

<h4><b>What is a classifier?</b></h4>
<p>A classifier lists the sequences and their relative abundance that constitute the fecal signature of a given source.</p>

<h4><b>I already processed my sequences using dada2, should I re-process everything using CUTADAPT and PEAR?</b></h4>
<p>Preliminary explorations showed that FORENSIC could not correctly identify the bacterial signatures in tested samples using data processed by dada2. This incompatibility might originate from the fact that dada2 may shorten sequences of a nucleotide during the denoising step. This is not an issue when studying the whole bacterial community, but since FORENSIC is looking for the exact sequence matches between the signatures and the sample to test, predictions can be significantly altered. </p>

<h4><b>Can I use data generated using another sequencing platform than Illumina?</b></h4>
<p>We only validated the predictions using Illumina sequence data. We would be pleased to test FORENSIC using other technologies, such as minION.</p>

<h4><b>FORENSIC tests automatically for multiples sources, is it possible to test samples for a specific classifier?</b></h4>
<p>The classifiers and the associated predictions are independent. The prediction of one classifier does not influence the predictions of the other classifiers.  In other words, identifying the sewage signature in a sample does not weaken the probability to detect the cow signature in that same sample.</p>

<h4><b>The source I would like to target is not in FORENSIC?</b></h4>
<p>If you have enough representative samples, we could implement this source in FORENSIC. Let's talk about it.</p>

<h4><b>How long should last an analysis?</b></h4>
<p>The analysis should take a few minutes, depending on the size of your fasta: If it takes longer than that, feel free to contact us.</p>
    </div>

            </div>
            <div class="col-md-1"></div>
        </div>
</body>
</html>
