<!DOCTYPE html>
<html>
    <head>
        <title>Microbial Source Tracker</title>
        <meta name="google" value="notranslate">
        <script src="https://code.jquery.com/jquery-3.1.1.min.js" integrity="sha256-hVVnYaiADRTO2PzUGmuLJr8BLUSjGIZsDYGmIJLv2b8=" crossorigin="anonymous"></script>
        <link href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous">
        <script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script>
        <link rel="stylesheet" type="text/css" href="{{ url_for('static', filename='css/style.css') }}">
        <link rel="stylesheet" type="text/css" href="{{ url_for('static', filename='css/lightbox.css') }}">
        <script src="{{ url_for('static', filename='js/jquery.form.js') }}" type="text/javascript"></script>
        <script src="{{ url_for('static', filename='js/index.js') }}" type="text/javascript"></script>
        <script src="{{ url_for('static', filename='js/lightbox.js') }}" type="text/javascript"></script>
        <link href="https://fonts.googleapis.com/css?family=Roboto&display=swap" rel="stylesheet">
        <style type="text/css">
            body {
                font-family: 'Roboto', sans-serif;
            }
        </style>
    </head>
    <body>
        <div class="container">
            <div class="jumbotron">
                <!-- <img src="{{ url_for('static', filename='img/animals.png') }}" width="240" height="240" style="float: right; display: inline-block;"> -->
                <a href="/"><img width="500px" src="{{ url_for('static', filename='img/logo.png') }}"></a>
                <p class="lead"></p>
                <div class="row" style="">
                    <div class="col-md-6">
                        <div class="panel panel-default">
                            <div class="panel-heading">
                                <h3 class="panel-title">Submit new</h3>
                            </div>
                            <div class="panel-body">
                                <form action="/newtask" method="post" enctype="multipart/form-data">
                                    <div class="form-group">
                                        <label class="control-label" for="name">Fasta file</label>
                                        <div class="input-group">
                                            <input type="text" class="form-control fileUploadTxt input-sm" readonly="" value="Fasta file (not compressed)">
                                            <label class="input-group-btn">
                                                <span class="btn btn-primary btn-sm">
                                                    Browse… <input type="file" style="display: none;" id="file" name="file">
                                                </span>
                                            </label>
                                        </div>
                                    </div>
                                    <div class="alert alert-info" style="padding: 5px; font-size: 13px;">
                                        <strong>Info!</strong> Headers have to contain the sample ID and the sequence ID separated by a vertical bar: <code>>SampleID|SequenceID</code>
                                    </div>
                                    <div class="form-group">
                                        <label class="control-label" for="email">E-mail</label>
                                        <input class="form-control" id="email" maxlength="100" name="email" placeholder="E-mail" title="" type="text">
                                    </div>
                                    <div class="form-group">
                                        <label for="region">16S region: </label>
                                        <select class="form-control" name="region" id="region">
                                            <option value="v4v5">v4</option>
                                            <option value="v6">v6</option>
                                        </select>
                                    </div>
                                    <div class="text-center">
                                        <a class="btn btn-primary btn-md" href="#" role="button" onclick="$('form').submit();"><span class="glyphicon glyphicon-upload"></span> Submit new task</a>
                                    </div>
                                </form>
                            </div>
                        </div>
                    </div>
                    <div class="col-md-6 text-center">
                        <div class="panel panel-default">
                            <div class="panel-heading">
                                <h3 class="panel-title">Check results</h3>
                            </div>
                            <div class="panel-body">
                                <div class="input-group">
                                    <input type="number" id="task_id" class="form-control" placeholder="Task Id">
                                    <span class="input-group-btn">
                                        <a class="btn btn-primary btn-md" href="#" id="view_results" role="button"><span class="glyphicon glyphicon-folder-open"></span> View results</a>
                                    </span>
                                </div>
                            </div>
                        </div>
                    </div>
                </div>
            </div>
            <div class="row">
                <div class="col-md-6">
                    <h3>Welcome to FORENSIC</h3>
                    <p style="text-align: justify">
                        FORest Enteric Source IdentifiCation is a web interface to identify fecal pollution source using random forest classification and 16S rRNA gene amplicons. It relies on the detection in environmental samples of source-specific fecal bacterial signatures characterized upstream. <br />
                        <br />
                        <h4>How to use FORENSIC</h4>
                        <div style="clear: both;"></div>
                        <span class="li-dot"> 1 </span>
                        <p style="padding-left: 25px;">Samples have to be sequenced by targeting the V4 or V6 hypervariable regions of the 16S rRNA gene using Illumina® platform. Primers used, and PCR conditions are described <a href="http://www.earthmicrobiome.org/protocols-and-standards/16s/">here</a> for V4 (Earth Microbiome V4 primers) and <a href="https://figshare.com/articles/16S_amplicon_fusion_primers_and_protocol_for_Illumina_platform_sequencing/833944">here</a> for V6.</p>
                         <span class="li-dot"> 2 </span>
                         <p style="padding-left: 25px;">Reads must be trimmed using <a href="https://cutadapt.readthedocs.io/en/stable/guide.html">CUTADAPT</a> and merged using <a href="https://sco.h-its.org/exelixis/web/software/pear/">PEAR</a> according to the V4 or V6 workflow (<a href="{{ url_for('static', filename='data/TrimMerge_v1.3.sh') }}">bash </a> or <a href="{{ url_for('static', filename='data/TrimMerge_v1.3.txt') }}">text</a> version). Warning: some parameters have to be set manually before being used as an automated script. This workflow will generate one fasta file containing your different samples.</p>
                         <span class="li-dot"> 3 </span>
                         <p style="padding-left: 25px;">FORENSIC supports only fasta files (not compressed). A header line is REQUIRED and consists of the sampleID and the sequence ID separated by a vertical bar "|" (if you followed the workflow above, you are good to go):<br>
                            >sampleID|sequenceID (example: >BeachSample1|D4ZHLFP1-44-D15PUACXX-2-1101-1236-2324)
                            <br> <br>
                            An example, representing the V6 Bacteroidales and Clostridiales assemblage in contaminated river samples (described in Roguet et al. in preparation), is downloadable <a href="{{ url_for('static', filename='data/test_v6_MKE.fa.gz') }}">here</a>. </p>
                    <br>
                    <br>
                    <br>
                    
                    <h4>Citation</h4>
                    <p style="text-align: justify">
                        If you use FORENSIC, please consider citing the article below<br />
                        Roguet A, Eren AM, Newton RJ and McLellan SL (2018). Fecal source identification using random forest. Microbiome, 6:185 <a href="https://microbiomejournal.biomedcentral.com/articles/10.1186/s40168-018-0568-3/">[DOI: 10.1186/s40168-018-0568-3]</a><br />
                        <br />
                        Roguet A, Esen ÖC, Newton RJ, Eren AM and McLellan SL (in preparation). FORENSIC, an online fecal source identification toolkit. Peer-Reviewed Journal<br />
                    </p>
                    </p>
                </div>
                <div class="col-md-6">
                    <div class="row" style="padding-bottom: 10px;">
                        <div class="col-md-6">
                            <a href="/result/176673677487" class="btn btn-success btn-lg btn-block btn-huge">Example report</a>
                        </div>
                        <div class="col-md-6">
                            <a href="/faq" class="btn btn-success btn-lg btn-block btn-huge">FAQ</a>
                        </div>
                    </div>
                      <br>
                      <br>
                      <br>
                      <br>
                      <br>
                    <h4>How it works</h4>
                    <p style="text-align: justify">
                        To identify fecal pollution sources, a classifier was first designed for each source using random forest classification. <br />
                        <br />
                        <div style="clear: both;"></div>
                        <span class="li-dot"> a </span>
                        <p style="padding-left: 25px;">A classifier is composed of host-specific and host-preferred sequences that are the most reliable to discriminate between sources. To identify these sequences, the samples of a given source (15-20 samples/source)were compared to the other sources. Classifiers were then trained with the sequences previously selected.</p>
                         <span class="li-dot"> b </span>
                         <p style="padding-left: 25px;">To classify an unknown sample, the sequences identical to the ones composing the classifiers are retained, and their relative abundance is calculated.</p>
                         <span class="li-dot"> c </span>
                         <p style="padding-left: 25px;">An unknown sample is considered to be contaminated by a source if the majority of these abundances are similar to the relative abundances of the sequences in the classifiers.</p>

                        <a href="{{ url_for('static', filename='img/figure01_big.png') }}" data-lightbox="image1"><img src="{{ url_for('static', filename='img/figure01_small.png') }}"/></a>
                        <br>
                        A detailed description of sequences processing, and random forest parameters are available in the paper <a href="https://microbiomejournal.biomedcentral.com/track/pdf/10.1186/s40168-018-0568-3">Roguet et al. 2018</a>.
                        <br>
                    </p>
                </div>
            </div>
        </div>
        <footer class="footer">
            <div class="container">
                <p class="text-muted"> Site developed by <a href="mailto:adelaide.roguet@gmail.com" class="person-social" target="_blank"><i class="fa fa-fw fa-envelope-square"></i>Adelaide Roguet</a> at <a href="https://people.sfs.uwm.edu/mclellanlab/" class="person-social" target="_blank"><i class="fa fa-fw fa-home"></i> McLellan lab</a> and <a href="mailto:ozcanesen@gmail.com" class="person-social" target="_blank"><i class="fa fa-fw fa-envelope-square"></i>Ozcan Esen</a> at <a href="http://merenlab.org/" class="person-social" target="_blank"><i class="fa fa-fw fa-home"></i> Meren lab</a> <br> <br>© 2019 Funded by National Institutes of Health<br>Not Peer Reviewed</p>
            </div>
        </footer>
    <div id="modalUploading" class="modal fade" data-backdrop="static" data-keyboard="false" tabindex="-1" role="dialog" aria-hidden="true" style="padding-top:15%; overflow-y:visible;">
        <div class="modal-dialog modal-m">
            <div class="modal-content">
                <div class="modal-header"></div>
                <div class="modal-body">
                    <div class="progress progress-striped active" style="margin-bottom:0;">
                        <div class="progress-bar" style="width: 100%"></div>
                    </div>
                </div>
            </div>
        </div>
    </div>
    <script>
        lightbox.option({
          'resizeDuration': 200,
          'wrapAround': true
        });
    </script>
    </body>
</html>
