<!DOCTYPE html>
<html>
<head>
    <title>Microbial Source Tracker</title>
    <meta name="google" value="notranslate">
    <script src="https://code.jquery.com/jquery-3.1.1.min.js" integrity="sha256-hVVnYaiADRTO2PzUGmuLJr8BLUSjGIZsDYGmIJLv2b8=" crossorigin="anonymous"></script>
    <link href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous">
    <script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/popper.js/1.12.9/umd/popper.min.js"></script>
    <link rel="stylesheet" type="text/css" href="{{ url_for('static', filename='css/style.css') }}">
    <link rel="stylesheet" href="{{ url_for('static', filename='css/examples.css') }}">
    <link rel="stylesheet" href="{{ url_for('static', filename='css/vizuly.css') }}">
    <link rel="stylesheet" href="{{ url_for('static', filename='css/vizuly_halo.css') }}">
    <script src="{{ url_for('static', filename='js/jquery.form.js') }}" type="text/javascript"></script>
    <script src="{{ url_for('static', filename='js/index.js') }}" type="text/javascript"></script>
    <script src="{{ url_for('static', filename='js/d3.min.js') }}" typ e="text/javascript"></script>
    <script src="{{ url_for('static', filename='js/vizuly_core.min.js') }}" type="text/javascript"></script>
    <script src="{{ url_for('static', filename='js/vizuly_halo.min.js') }}" type="text/javascript"></script>
    <script src="{{ url_for('static', filename='js/halo_test_AR.js') }}" type="text/javascript"></script>
    <script src="{{ url_for('static', filename='js/braycurtis_heatmap.js') }}" type="text/javascript"></script>
    <link href="https://fonts.googleapis.com/css?family=Roboto&display=swap" rel="stylesheet">
    <style type="text/css">
        body {
            font-family: 'Roboto', sans-serif;
        }
    </style>
    <style>
        body {
          -webkit-print-color-adjust: exact !important;
        }
        .chart-item {
            display: inline-block;
            width: 20px;
            height: 20px;
            border: 1px solid rgba(0,0,0,0.1);
            border-radius: 10px;
            box-shadow: 2px 2px rgba(0,0,0,0.1);
        }

        .chart-item:hover {
            border: 1px solid rgba(255,255,255,0.8);
        }

        .legend td {
            padding: 2px 10px;
        }

        .borderless td, .borderless th {
            border-top: none !important;
        }

    </style>
</head>

<body>
   <body>
        <div class="container">
            <div class="row">
                <div class="col-md-12" style="text-align: center;">
                    <h2><a href="/"><img width="500px" src="{{ url_for('static', filename='img/logo.png') }}"></a><br>Report</h2>
                </div>
            </div>
            {% if data.exists %}
                {% if data.successful %}
                <div class="row">
                    <div class="col-md-3 col-md-offset-1">
                        <div class="panel panel-default">
                          <div class="panel-heading">Task details</div>
                          <div class="panel-body">
                            <table class="table borderless">
                                <tr>
                                    <td>Sender:</td>
                                    <td>{{ data.email }}</td>
                                </tr>
                                <tr>
                                    <td>Submitted:</td>
                                    <td>{{ data.submitted_date }}</td>
                                </tr>
				<tr>
                                    <td>Task ID:</td>
                                    <td>{{ data.task_id }}</td>
                                </tr>
                                <tr>
                                    <td>Total sample:</td>
                                    <td>{{ data.Clostridiales|length - 1 }}</td>
                                </tr>
                                <tr>
                                    <td>16S rRNA region detected:</td>
                                    <td>{{ data.region }}</td>
                                </tr>
                            </table>
                            <div style="text-align: center">
                              <a class="btn btn-primary" href="/result/{{ data.task_id }}/download" role="button"><span class="glyphicon glyphicon-download"></span> Download report</a>
                              </button>
                              <button type="button" class="btn btn-success" onclick="window.print();">
                                <span class="glyphicon glyphicon-print"></span> Print
                              </button>
                            </div>
                          </div>
                        </div>
                        <a href="/classifier" class="btn btn-success btn-lg btn-block btn-huge">Classifier Information</a>
			<a href="/faq" class="btn btn-success btn-lg btn-block btn-huge">FAQ</a>

                    </div>
                    <div class="col-md-7">
                        <div class="alert alert-info" role="alert">
                            <p>The fasta submitted has been processed through both <i><b>Bacteroidales</b></i> and <i><b>Clostridiales</b></i> classifiers. Source fecal predictions generated by random forest and bacterial assemblage characterizations are displayed below through interactive interfaces.
				<br>
				<br>
				FORENSIC interactive interface performs optimally using Chrome. Safari, Opera and Edge work fine. However, please consider using another browser if your default one is Firefox or Internet Explorer, which do not display the results correctly.</p></div>
                            
                        <div class="panel panel-default">
                          <div class="panel-heading">Fecal Source Predictions</div>
                          <div class="panel-body">
                              <table class="table">
                                {% set classifiers = ['Clostridiales', 'Bacteroidales'] %}
                                <tr>
                                    <td style="text-align: center">
                                        Sample Name
                                    </td>
                                    <td style="text-align: center">
                                        Classifier
                                    </td>
                                    {% for animal in data.animals %}
                                    <td style="text-align: center">
                                        {{ animal }}
                                    </td>
                                    {% endfor %}
                                    <td style="text-align: center">
                                        Reads matched (unique reads)
                                    </td>
                                    <td style="text-align: center">
                                        Total reads
                                    </td>
                                    <td style="text-align: center">
                                        Percentage
                                    </td>
                                </tr>
                                {% for sample in data.sample_names %}
                                    {% set sample_loop = loop %}
                                    {% for classifier in classifiers %}
                                    {% set classifier_loop = loop %}
                                    <tr style="{% if classifier_loop.index % 2 == 1 %}border-top: 2px solid #888787;{% endif %}">
                                        {% if classifier_loop.index % 2 == 1 %}
                                        <td style="text-align: center" rowspan="2">
                                            {{ sample }}
                                        </td>
                                        {% endif %}
                                        <td style="text-align: center">
                                            {{ classifier }}
                                        </td>
                                        {% set classifier_animal = data[classifier][0][loop.index-1] %}
                                        {% for item in data[classifier][sample_loop.index][1:] %}
                                        <td style="text-align: center">
                                            <div class="chart-item" style="background-color: 
                                            {% if item|float() == 0|float() %}
                                                #ffffff
					    {% elif data['classifier_type'][data.region][data[classifier][0][loop.index-1]][classifier] == 'discarded' %}
                                                black
                                            {% elif item|float() < data['cutoff_low'][data.region][data[classifier][0][loop.index-1]][classifier]|float()  %}
                                                #DBFAD4
                                            {% elif item|float() < data['cutoff'][data.region][data[classifier][0][loop.index-1]][classifier]|float()  %}
                                                #60D836
                                            {% else %}
                                                #9E0142
                                            {% endif %}
                                            !important;
                                            {% if data['classifier_type'][data.region][data[classifier][0][loop.index-1]][classifier] == 'global' %}
						border: 2px solid black 
					   {% elif data['classifier_type'][data.region][data[classifier][0][loop.index-1]][classifier] == 'draft' %}
						border: 2px solid grey
						!important;{% endif %}"  data-toggle="tooltip" data-placement="top" title="{{ item }}">
						</div>
                                        </td>
                                        {% endfor %}
                                        <td style="text-align: center">
                                            {{ data[classifier + '_samples_sums'][sample] }} ({{ data[classifier + '_samples_unique_count'][sample] }})
                                            
					    {% if data[classifier + '_samples_unique_count'][sample] == 0 %}							  
                                            <a href="#" data-toggle="tooltip" title="No sequence was matching to the classifier. Did you encounter issues during the trimming step? Was the sequencing depth enough?">
                                            <span class="glyphicon glyphicon-exclamation-sign"></span></a>
                                            {% endif %}
				      
                                            {% if data[classifier + '_samples_unique_count'][sample] > 0 and data[classifier + '_samples_unique_count'][sample] < 100 or data[classifier + '_samples_sums'][sample] < 1000 %}
                                            <a href="#" data-toggle="tooltip" title="Too few sequences in your sample matching to the classifiers (<1,000 reads matched and/or <100 unique reads), it can affect the predictions.">
                                            <span class="glyphicon glyphicon-exclamation-sign"></span></a>
                                            {% endif %}
                                        </td>
                                        <td style="text-align: center">
                                            {{ "{:,d}".format(data['reads_per_sample'][sample])}}

                                            {% if data['reads_per_sample'][sample] < 5000 %}
                                            <a href="#" data-toggle="tooltip" title="Too few reads in this sample, it can affect the predictions.">
                                            <span class="glyphicon glyphicon-exclamation-sign"></span></a>
                                            {% endif %}
                                        </td>
                                        <td style="text-align: center">
                                            {{ "{0:.2f}".format(100 * data[classifier + '_samples_sums'][sample]/data['reads_per_sample'][sample])}}%
                                        </td>
                                    </tr>
                                    {% endfor %}
                                {% endfor %}
                                </table>
                                <div style="float: right;">
                                    <table class="legend">
					<tr>
                                            <td rowspan="2">Fecal signature detection confidence index</td>
                                            <td>not detected</td>
                                            <td>low confidence (&lt;80% specificity)</td>
                                            <td>low confidence (&ge;80% specificity)</td>
                                            <td>high confidence </td>
                                            <td></td>
                                        </tr>
                                        <tr>
                                            <td><div class="chart-item" style="background-color: #ffffff !important;"></div></td>
                                            <td><div class="chart-item" style="background-color: #DBFAD4 !important;"></div></td>
                                            <td><div class="chart-item" style="background-color: #60D836 !important;"></div></td>
                                            <td><div class="chart-item" style="background-color: #9E0142 !important;"></div></td>
                                            <td></td>
                                        </tr>
                                    </table>
                                </div>
			    <div style="float: left;">
                                    <table class="legend">
                                        <tr>
                                            <td rowspan="2">Classifier types</td>
                                            <td>global</td>
                                            <td>draft</td>
                                            <td>discarded</td>
                                            <td></td>
                                        </tr>
                                        <tr>
                                            <td><div class="chart-item" style="border: 2px solid black !important"; border: 2px solid black !important;"></div></td>
                                            <td><div class="chart-item" style="border: 2px solid grey !important;"></div></td>
                                            <td><div class="chart-item" style="background-color: black !important;"></div></td>
                                            <td></td>
                                            <td></td>
                                        </tr>
                                    </table>
                                </div>
                                <div class="clearfix"></div>
                                <button onclick="$('#legend1_1').toggle();" class="btn btn-info" style="margin-bottom: 10px;">Show Info</button>
                                <div id="legend1_1" class="panel panel-default" style="display: none; text-align: justify;">
                                    <div class="panel-body">
					    <p>How is this value obtained? Submitted bacterial profiles are mapped along with decision trees previously built by random forest to train the classifiers. Each tree gives a classification (roughly: yes/no the mapped data is/is not close to the model). Combing all voting tree classifications provides the voting tree probability. Thus, the voting tree probability translates the degree of similarity between the signatures and the unknown samples. The more similar the bacterial signature of each sample is to the set classifier fecal signatures, the higher the probability is in the source prediction output.</p>
					    <br>
					    <p>What is the difference between "low confidence" and "high confidence" regarding the fecal signature detection confidence index? We optimized for each classifier the voting tree cut-off (decision cut-off) at which a sample is considered as contaminated (or not). This cut-off was defined to maximize the sensitivity and the specificity of the classifiers. Above the decision cut-off, a sample is classified as "high confidence", i.e., the fecal signature has been detected in the submitted sample. Below the decision cut-off, a sample is classified as "low confidence", i.e., a fraction of the fecal signature was recovered in the submitted sample, but not enough to be classified as contaminated. </p>
					    <p>"&ge;80% specificity" indicates that at the voting tree probability observed in the submitted sample, the classifier had a specificity of at least 80% (evaluated using raw fecal samples). </p>
					    <br>
					    <p>Please consult our <a href="/faq" >FAQ</a> to have more details about the global, draft (and excluded) classifiers.</p>
                                    </div>
                                </div>
                          </div>
                        </div>
	                  <div class="panel panel-default">
                          <div class="panel-heading">Fecal Signature Characterization</div>
                          <div class="panel-body">
                            Sample Name: <select id="bubbleplot_sample_name">
                                {% for sample_name in data.sample_names %}
                                    <option val="clostridiales_{{sample_name}}">clostridiales_{{sample_name}}</option>
                                {% endfor %}
                                {% for sample_name in data.sample_names %}
                                    <option val="bacteroidales_{{sample_name}}">bacteroidales_{{sample_name}}</option>
                                {% endfor %}
                            </select>
                            <li id="currentDisplay" class="selected" style="display: none;"><a></a></li>
                            <div id="viz_container">

                            </div>
                            <div class="clearfix"></div>
                            <button onclick="$('#legend3').toggle();" class="btn btn-info" style="margin-bottom: 10px;">Show Info</button>
                            <div id="legend3" class="panel panel-default" style="display: none; text-align: justify;">
                                <div class="panel-body">
                                <p>Bubbles represent the amplicon sequence variants (ASVs) that compose the classifiers (from all sources). Blue bubbles represent the ASVs that matched with the sample submitted. Red bubbles represent the ASVs that were not found in the sample submitted. The bubble size is proportional to the relative abundance of the ASV in the sample (all the ASVs in a submitted sample scale to 100% for that specific analysis). Bubbles that are associated with several source classifiers have a darker inner circle.</p>
					<p>Each arc of the outer circle represents a source. Longer arcs indicate a particular source contributed more to the fecal pollution (among the sources investigated).</p> 
					<p>Each ray along the circumference represents the relative abundance of the ASV within a source.</p>
                                <br>
                                <img width="700px" src="{{ url_for('static', filename='img/info_bubbleplot.png') }}">
                                </div>
                            </div>
                          </div>
                        </div>
                    </div>
                </div>
                {% else %}
                <div class="row">
                    <div class="col-md-12">
                        <h4>Classifier failed for this FASTA file, it could not find any match.</h4>
                    </div>
                </div>
                {% endif %}
            {% else %}
            <div class="row">
                <div class="col-md-12">
                    <h4>Task Id not found</h4>
                </div>
            </div>
            {% endif %}
        </div>
<script type="text/javascript">

$(document).ready(function() {
    $(function () {
      $('[data-toggle="tooltip"]').tooltip()
    });
});

    //Once the document is ready we set javascript and page settings
    var screenWidth;
    var screenHeight;
    var task_id = '{{data.task_id}}';

    $(document).ready(function () {

        var rect;
        if (self==top) {
            rect = document.body.getBoundingClientRect();
        }
        else {
            rect =  parent.document.body.getBoundingClientRect();
        }

        //Set display size based on window size.
        screenWidth = (rect.width < 960) ? Math.round(rect.width*.95) : Math.round((rect.width - 210) *.95)
        screenHeight = Math.min(parent.innerHeight * 0.75, screenWidth);
        screenWidth = screenHeight;

        d3.select("#currentDisplay")
                .attr("item_value", screenWidth + "," + screenHeight)
                .attr("class", "selected")
                .html("<a>" + screenWidth + "px - " + screenHeight + "px</a>");

        // Set the size of our container element.
        viz_container = d3.selectAll("#viz_container")
                .style("width", screenWidth + "px")
                .style("height", screenHeight + "px");

        $('#bubbleplot_sample_name').change(function() {
             $('#viz_container').empty();
	     loadData(task_id, $('#bubbleplot_sample_name').val());
        });
        $('#bubbleplot_sample_name').trigger('change');


        drawHeatmap({{data.task_id}}, '#heatmap'); // heatmap.js

    });

</script>
</body>
</html>
